app/controllers/application_controller.rb

```diff
@@ -78,47 +64,31 @@ class Base < Sinatra::Base
       settings.http_allow_headers.join(',')
   end
 
-  get '/ping' do
-    { 
-      status: 'Ok'
-    }.to_json
-  end
-
-  get '/jobs' do
-    jobs = []
-    etcd.get('/queue', recursive: true).children.each do |job|
-      job = JSON.parse(job.value)
-      jobs << JobPresenter.single(job) if job['created_from'] == 'API'
-    end
-    jobs.to_json
+  before do
+    Tendrl.etcd = Etcd.client(
+      host: settings.etcd_config[:host],
+      port: settings.etcd_config[:port],
+      user_name: settings.etcd_config[:user_name],
+      password: settings.etcd_config[:password]
+    )
   end
 
-  get '/jobs/:job_id' do
-    jobs = []
-    job = JSON.parse(etcd.get("/queue/#{params[:job_id]}").value)
-    JobPresenter.single(job).to_json
-  end
+  protected
 
-  get '/jobs/:job_id/logs' do
-    params[:type] ||= 'all'
-    job = JSON.parse(etcd.get("/queue/#{params[:job_id]}").value)
-    request_id = job['request_id']
-    logs = etcd.get("/#{request_id}/#{params[:type]}").value
-    { logs: logs, type: params[:type] }.to_json
+  def username
+    request.user_agent
```

Comment:
I understand [this](https://developer.github.com/v3/#user-agent-required) text about User-Agent header like: 

> We don't accept request with empty User-Agent header. If you use something different than browser(which sends this header automatically) you should add non-empty User-Agent header. 

I think it does not mean that github uses this header for login name ID. Please consider that if user uses API function from browser (which is btw. typical use-case for Tendrl UI) it should be identified in different browser instances of same version (which is not possible with this proposal). Also I would like to point out that if you use User-Agent header for user ID user should change this browser identification everytime he would log in Tendrl UI even if you would like to login as different user because you would like to act with different role in Tendrl.
Also we would like to have login which can be remembered by humans and  not something like ```Mozilla/5.0 (X11; Linux x86_64; rv:51.0) Gecko/20100101 Firefox/51.0``` which is User-Agent of my browser.